std/asys/FileWatcher.hx

```diff
@@ -0,0 +1,47 @@
+package asys;
+
+import haxe.Error;
+import haxe.NoData;
+import haxe.async.*;
+import haxe.io.FilePath;
+
+/**
+	File watchers can be obtained with the `asys.FileSystem.watch` method.
+	Instances of this class will emit signals whenever any file in their watched
+	path is modified.
+**/
+class FileWatcher {
```

Comment:
Shouldn't this be `extern`?

Reply:
No, it has logic and wraps `FileWatcherNative`, which is the extern. That's the general pattern in asys.

Reply:
But it has constructor, which accepts arguments, but does not handle them. Also `close` method doesn't really close anything.

Reply:
Oh, I was looking at the one from #8817. Maybe splitting up the PRs wasn't such a good idea after all...

Reply:
When splitting this from 8817 I removed any uses of the "native" handles, either by making the function `extern`, or removing (most) of the body. This PR is primarily about the API, less about the implementations.